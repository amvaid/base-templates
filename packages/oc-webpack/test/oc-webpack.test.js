jasmine.DEFAULT_TIMEOUT_INTERVAL = 30000;

('use strict');

const path = require('path');
const api = require('../index.js');
const webpackConfigurator = require('../lib/configurator');
const webpackCompiler = require('../lib/compiler');

test('module APIs', () => {
  expect(api).toMatchSnapshot();
});

test('webpack configurator', () => {
  const config = webpackConfigurator({
    stats: 'errors-only',
    dependencies: {},
    publishFileName: 'server.js',
    serverPath: '/path/to/server.js'
  });

  // clean paths
  const target = config.module.rules[0].use;
  target[0].loader = path.relative(__dirname, target[0].loader);
  target[1].loader = path.relative(__dirname, target[1].loader);
  target[1].options.presets[0][0] = path.relative(
    __dirname,
    target[1].options.presets[0][0]
  );
  expect(config).toMatchSnapshot();
});

test('webpack compiler', done => {
  const config = webpackConfigurator({
    stats: 'errors-only',
    dependencies: {},
    publishFileName: 'server.js',
    serverPath: path.join(
      __dirname,
      '../../../mocks/jade-component',
      'server.js'
    )
  });
  webpackCompiler(config, (warning, serverContentBundled) => {
    expect(serverContentBundled).toMatchSnapshot();
    done();
  });
});
