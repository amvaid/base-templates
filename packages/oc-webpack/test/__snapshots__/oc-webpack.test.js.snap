// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module APIs 1`] = `
Object {
  "compiler": [Function],
  "configurator": [Function],
  "webpack": [Function],
}
`;

exports[`webpack compiler 1`] = `"module.exports=function(a){function b(d){if(c[d])return c[d].exports;var e=c[d]={i:d,l:!1,exports:{}};return a[d].call(e.exports,e,e.exports,b),e.l=!0,e.exports}var c={};return b.m=a,b.c=c,b.d=function(a,c,d){b.o(a,c)||Object.defineProperty(a,c,{configurable:!1,enumerable:!0,get:d})},b.n=function(a){var c=a&&a.__esModule?function(){return a['default']}:function(){return a};return b.d(c,'a',c),c},b.o=function(a,b){return Object.prototype.hasOwnProperty.call(a,b)},b.p='',b(b.s=0)}([function(a,b,c){'use strict';Object.defineProperty(b,'__esModule',{value:!0});const d=c(1);b.data=(a,b)=>b(null,{})},function(a){a.exports=require('lodash')}]);"`;

exports[`webpack compiler verbose 1`] = `"module.exports=function(a){function b(d){if(c[d])return c[d].exports;var e=c[d]={i:d,l:!1,exports:{}};return a[d].call(e.exports,e,e.exports,b),e.l=!0,e.exports}var c={};return b.m=a,b.c=c,b.d=function(a,c,d){b.o(a,c)||Object.defineProperty(a,c,{configurable:!1,enumerable:!0,get:d})},b.n=function(a){var c=a&&a.__esModule?function(){return a['default']}:function(){return a};return b.d(c,'a',c),c},b.o=function(a,b){return Object.prototype.hasOwnProperty.call(a,b)},b.p='',b(b.s=0)}([function(a,b,c){'use strict';Object.defineProperty(b,'__esModule',{value:!0});const d=c(1);b.data=(a,b)=>b(null,{})},function(a){a.exports=require('lodash')}]);"`;

exports[`webpack compiler with fatal error 1`] = `"This is a fatal compilation error"`;

exports[`webpack configurator 1`] = `
Object {
  "entry": "/path/to/server.js",
  "externals": Array [
    [Function],
    /\\^\\[a-z@\\]\\[a-z\\\\-\\\\/0-9\\\\\\.\\]\\+\\$/i,
  ],
  "module": Object {
    "rules": Array [
      Object {
        "exclude": /node_modules/,
        "test": /\\\\\\.js\\$/,
        "use": Array [
          Object {
            "loader": "../../infinite-loop-loader/index.js",
          },
          Object {
            "loader": "../../../node_modules/babel-loader/lib/index.js",
            "options": Object {
              "babelrc": false,
              "cacheDirectory": true,
              "plugins": Array [
                Array [
                  "../../../node_modules/babel-plugin-transform-object-rest-spread/lib/index.js",
                ],
              ],
              "presets": Array [
                Array [
                  "../../../node_modules/babel-preset-env/lib/index.js",
                  Object {
                    "modules": false,
                    "targets": Object {
                      "node": 6,
                    },
                  },
                ],
              ],
            },
          },
        ],
      },
    ],
  },
  "output": Object {
    "filename": "server.js",
    "libraryTarget": "commonjs2",
    "path": "/build",
  },
  "plugins": Array [
    BabelMinifyPlugin {
      "options": Object {
        "babel": Object {
          "transform": [Function],
        },
        "comments": /\\^\\\\\\*\\*!\\|@preserve\\|@license\\|@cc_on/,
        "jsregex": /\\\\\\.js\\(\\$\\|\\\\\\?\\)/i,
        "minifyOpts": Object {},
        "minifyPreset": [Function],
        "parserOpts": Object {},
        "sourceMap": undefined,
      },
    },
    DefinePlugin {
      "definitions": Object {
        "process.env.NODE_ENV": "\\"production\\"",
      },
    },
  ],
  "stats": "errors-only",
  "target": "node",
}
`;
